### ISSUE TITLE
`:` instead of `->`

### ISSUE THREAD

**Issue by jgarte** [ruricolist/serapeum#177]
Hi,

What do you think if instead of `->` for annotating types that we use `:`?

So for example,

```lisp
(: dict* (hash-table &rest t) hash-table)
(defun dict* (dict &rest args)
  ...)
```

Instead of

```lisp
(-> dict* (hash-table &rest t) hash-table)
(defun dict* (dict &rest args)
  ...)
```
What do you think?

**Comment by mdbergmann** [comment]
Yes, this wouldn't clash with `binding-arrows`, `arrows` and those things.

**Comment by ruricolist** [comment]
I like it aesthetically, but I don't think it's workable in CL. It would have to be written `:||`; not all Lisps allow binding macros to keywords; and generally it would be considered bad style in CL for a library to change a global binding outside its package.

**Comment by jgarte** [comment]
Thanks for the reply. Your points make sense. Not sure what else to add but thanks for the information. I'll ponder it further. I might just write vanilla type specifiers instead when needed since the arrow conflicts with my threading arrow usually.
